### PR TITLE
The Button component

### PR DESCRIPTION
**Pull request by daoanhhuy26012001**
- [x] Define the `Button` component
- [x] Set styles

### CODE REVIEW

File: src/components/common/Button/index.tsx [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+// Styles
+import './button.css';
+
+interface ButtonProps extends ButtonHTMLAttributes<HTMLButtonElement> {
+  variant?: string;
+  children?: ReactNode;
+  onClick?: MouseEventHandler<HTMLButtonElement>;
+}
+
+const Button = ({ variant = '', children = '', onClick }: ButtonProps) => {
```
Comment (TungLam101098) [inline src/components/common/Button/index.tsx]: default onClick?

File: src/components/common/Button/index.tsx [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  switch (variant) {
+  case 'primary':
+    buttonClassName = 'button-primary';
+    break;
+  case 'secondary':
+    buttonClassName = 'button-secondary';
+    break;
+  default:
+    buttonClassName = 'button-tertiary';
+  }
```
Comment (TungLam101098) [inline src/components/common/Button/index.tsx]: You should not use like this.
we can use `className={button button-${variant}}`

File: src/components/common/Button/index.tsx [hunk 3]
```diff
@@ -0,0 +1,33 @@
+import { ButtonHTMLAttributes, MouseEventHandler, ReactNode } from 'react';
+
+// Styles
+import './button.css';
+
+interface ButtonProps extends ButtonHTMLAttributes<HTMLButtonElement> {
+  variant?: string;
```
Comment (TungLam101098) [inline src/components/common/Button/index.tsx]: variant should type: primary, secondary, tertiary

Comment (daoanhhuy26012001) [reply src/components/common/Button/index.tsx]: Done

File: src/components/common/Button/index.tsx (hunk 2 above)
Comment (daoanhhuy26012001) [reply src/components/common/Button/index.tsx]: Done

File: src/components/common/Button/index.tsx (hunk 1 above)
Comment (daoanhhuy26012001) [reply src/components/common/Button/index.tsx]: Done

File: src/components/common/Button/index.tsx [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+// Styles
+import './button.css';
+
+interface ButtonProps extends ButtonHTMLAttributes<HTMLButtonElement> {
+  variant?: 'primary' | 'secondary' | 'tertiary';
+  children?: ReactNode;
+  onClick?: MouseEventHandler<HTMLButtonElement>;
+}
+
+const Button = ({
```
Comment (TungLam101098) [inline src/components/common/Button/index.tsx]: We can use Button.defaultProps = {
variant = 'primary',
  children = '',
  onClick = () => {}
}

File: src/components/common/Button/index.tsx [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+  variant = 'primary',
+  children = '',
+  onClick = () => {}
+}: ButtonProps) => {
+  return (
+    <button className={`button button-${variant}`} onClick={onClick}>
+      {children}
+    </button>
+  );
```
Comment (TungLam101098) [inline src/components/common/Button/index.tsx]: Should use shorthand

Comment (daoanhhuy26012001) [reply src/components/common/Button/index.tsx]: Done

**Pull request merged**